docs/readmes/proposals/p010_resource-tagging.md

```diff
@@ -0,0 +1,191 @@
+are used by existing deployments. This feature will not be backported and will
+be supported from the Magma release it lands, tentatively 1.5.
+
+
+## Observability and Debug
+
+All tags are retrievable using CLI options (aws, kubectl) and validated. The
+respective CLIs can also be used to filter resources based on tags.
+
+
+## Implementation
+
+
+
+*   Proposal review and inviting comments on Github discussions
+*   Implementation plan for AWS and Kubernetes resources as the first
+    deliverable to support currently supported deployment models (AWS or
+On-Prem) for v1.5 targeted for April 2021.
+*   Phase 1 will implement tagging for all AWS resources
+*   Phase 2 will tag all other kubernetes resources in orchestrator
+
+
+## Future Work
+
+
+
+*   Most Cloud provider environments provide a way to tag resources. The current
+    proposal will implement the findings on AWS and will extend the support to
+other Cloud provider platforms and on-prem options. However, as new services are
+containerized and managed via kubernetes, they can be tagged even if they are
+outside the scope of the orchestrator (ex: Cloud HA instances)
+
+
+## Design
+
+
+
+*   For tags, follow the current guidelines
+    *   Minimal number of tags to start with and incrementally increasing as
+        additional use-cases are identified
+	*   The primary use-case for now is to use labels and tags for A/B or
+	    Blue/Green or Canary based update of services
+	*   The secondary use-case for now is to associate all resources in a
+	    deployment together so that they can be searched for and cleaned up.
+    *   Identify mandatory, optional and resource-specific tags 
+	*   Mandatory: UUID, named magmaUUID
+    *   For Cloud deployments, Terraform is currently used with a provider for
+        the actual Cloud platform that hosts the services. Provider-native
+mechanisms will be used to tag the resource. An example block is given below
+where an AWS provider resource is tagged with the UUID.
+
+	``` resource "aws_db_instance" "nms" { identifier        =
+var.nms_db_identifier allocated_storage = var.nms_db_storage_gb engine
+= "mysql" engine_version    = var.nms_db_engine_version instance_class    =
+var.nms_db_instance_class
+
+  Tags = { magmaUUID = "<128-bit Identifier>" } } ```
+
+
+*   Any non-Cloud provider resources will be deployed via Kubernetes and can be
+    tagged (labeled) as follows. These resources can be filtered via label
+selectors via the kubectl command. For instances starting up, use the kubernetes
+configuration file. To generate dynamic configuration manifests, extensions are
+available to create custom kubernetes operators in the future.
+
+    ``` template: metadata: labels: app.kubernetes.io/component: orc8r
+app.kubernetes.io/instance: <Identifier> ```
+
+
+*   For instances already running, use the kubectl label call to set it during
+    runtime
+
+<table> <tr> <td> <code>            kubectl label pods &lt;orc8r pods>
```

Comment:
Inline HTML looks suss. Is this required or can we accomplish this another way

Reply:
G Docs -> Code Blocks plugin -> Markdown plugin -> vi == That. Fixed.